### PR TITLE
[DE-902] Refactor social share

### PR DESCRIPTION
**Pull request by Cromeror**

### CODE REVIEW

File: src/customJs/socialShareTool/configuration.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    socialImg:
+      'https://app2.dopplerfiles.com/MSEditor/images/color_big_pinterest_en.png',
+    altText: 'Pinterest',
+    id: 'pinterest',
+  },
+  {
+    link: '[[shareInWhatsappLink]]',
+    socialImg:
+      'https://app2.dopplerfiles.com/MSEditor/images/color_big_whatsapp_en.png',
+    altText: 'Whatsapp',
```
Comment (Cromeror) [inline src/customJs/socialShareTool/configuration.js:34]: Quizá tiene mas sentido usar un nombre diferente.
```suggestion
    name: 'Whatsapp',
```

File: src/customJs/socialShareTool/enableSocialProperty.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+      <label className="blockbuilder-label-primary">
+        <div className="">
+          <span className="has-value same-value">{label}</span>
+        </div>
+      </label>
+      <label className="blockbuilder-label-right">
+        <div
+          className={`react-toggle ${active ? 'react-toggle--checked' : ''}`}
+          onClick={clickEvent}
+        >
```
Comment (andresmoschini) [inline src/customJs/socialShareTool/enableSocialProperty.js:28]: Is it keyboard friendly?

File: src/customJs/socialShareTool/enableSocialProperty.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
+import PropTypes from 'prop-types';
+const React = window.unlayer.React;
+
+const Toggle = ({ label, onChange }) => {
+  const [active, setActive] = React.useState(true);
+  const clickEvent = (e) => {
+    e.preventDefault();
```
Comment (andresmoschini) [inline src/customJs/socialShareTool/enableSocialProperty.js:7]: What are you preventing here?

File: src/customJs/socialShareTool/enableSocialProperty.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    updateValue(currentSelection.filter((id) => id !== key));
+  };
+
+  return options.map((option, index) => {
+    return (
+      <Toggle
+        key={index}
+        onChange={(value) => changeValue(value, option.id)}
+        label={option.name}
+      />
```
Comment (andresmoschini) [inline src/customJs/socialShareTool/enableSocialProperty.js:75]: It is not possible to set the toggle state from the outside.

File: src/customJs/socialShareTool/enableSocialProperty.js (hunk 3 above)
Comment (andresmoschini) [reply src/customJs/socialShareTool/enableSocialProperty.js:7]: I did not understand, what is the default handler, I mean, what are you preventing?

Comment (Cromeror) [reply src/customJs/socialShareTool/enableSocialProperty.js:8]: I correct, I need e.preventDefault to allow clicking on the toggle. The way more easily was attach the onclick Behavior over the toggle parent.

File: src/customJs/socialShareTool/enableSocialProperty.js (hunk 2 above)
Comment (Cromeror) [reply src/customJs/socialShareTool/enableSocialProperty.js]: Fixed now.

File: src/customJs/socialShareTool/enableSocialProperty.js (hunk 4 above)
Comment (Cromeror) [reply src/customJs/socialShareTool/enableSocialProperty.js]: I did think about that and I prefer keep internal state in the Toggle component, maybe in the future we could add to the Toggles api a new prop named Value

File: src/customJs/socialShareTool/enableSocialProperty.js (hunk 3 above)
Comment (andresmoschini) [reply src/customJs/socialShareTool/enableSocialProperty.js:8]: I did not understand yet, we can clarify it later in a call. Thanks!

File: src/customJs/socialShareTool/enableSocialProperty.js (hunk 2 above)
Comment (andresmoschini) [reply src/customJs/socialShareTool/enableSocialProperty.js]: great!

**Comment by andresmoschini** [comment]
Great work!

**Pull request merged**